### PR TITLE
Vending Machine Launch Fix

### PR DESCRIPTION
**Pull request by Ben10083**
Fixes https://github.com/Aurorastation/Aurora.3/issues/20201

Fixes runtime in SSThrowing by having non-mobs default to targeting chest
Refactors vendor reply code to be similar to how it handles ads and slogans
Fixes interaction involving using a string to get the coin back from vending machine

### CODE REVIEW

**Comment by Ben10083** [comment]
!review

**Comment by Ben10083** [comment]
!bugfix

**Review by FluffyGhoster — DISMISSED** [review]
This isn't an acceptable fix, the parameter of this proc types that var as a mob, if the caller isn't respecting the proc signature it's an issue of what calls it, not of the proc

**Comment by Ben10083** [comment]
> This isn't an acceptable fix, the parameter of this proc types that var as a mob, if the caller isn't respecting the proc signature it's an issue of what calls it, not of the proc

Unsure how to resolve in a different way, contacting you over discord

**Comment by Ben10083** [comment]
Going to expand this regarding some refactoring and other stuff I want to do to vendor while looking

File: code/game/machinery/vending.dm [hunk 1]
```diff
@@ -698,8 +745,9 @@
 		R.amount--
 		SStgui.update_uis(src)
 
-	if(((src.last_reply + (src.vend_delay + 200)) <= world.time) && src.vend_reply)
-		src.speak(src.vend_reply)
+	if(((src.last_reply + (src.vend_delay + 200)) <= world.time) && (src.reply_list.len > 0) && !src.shut_up)
```
Comment (FluffyGhoster) [inline code/game/machinery/vending.dm:748]:
```suggestion
	if(!src.shut_up && ((src.last_reply + (src.vend_delay + 200)) <= world.time) && length(src.reply_list))
```

File: code/game/machinery/vending.dm [hunk 2]
```diff
@@ -698,8 +745,9 @@
...
+		var/reply = pick(src.reply_list)
+		src.speak(reply)
```
Comment (FluffyGhoster) [inline code/game/machinery/vending.dm:750]:
```suggestion
		src.speak(pick(src.reply_list))
```

File: code/game/machinery/vending.dm [hunk 3]
```diff
@@ -679,11 +724,13 @@
...
+				var/obj/item/coin/returned_coin = new(coin)
+				returned_coin.AddOverlays("coin_string_overlay")
+				returned_coin.string_attached = TRUE
+				user.put_in_hands(returned_coin)
 			else
 				to_chat(user, SPAN_WARNING("You weren't able to pull the coin out fast enough, and the machine ate it!"))
-				QDEL_NULL(coin)
-		else
-			QDEL_NULL(coin)
+		QDEL_NULL(coin)
```
Comment (FluffyGhoster) [inline code/game/machinery/vending.dm:733]: No, don't recreate the coin, just put it back to the person

**Review by hazelrat — APPROVED** [review]

File: code/game/machinery/vending.dm [hunk 4]
```diff
@@ -677,13 +722,14 @@
 			return
 
 		if(coin.string_attached)
-			if(prob(50))
+			if(prob(90))
```
Comment (alsoandanswer) [inline code/game/machinery/vending.dm:725]:
```suggestion
			if(prob(50))
```

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**